gobblin-core/src/test/java/gobblin/source/extractor/extract/jdbc/JdbcExtractorTest.java

```diff
@@ -0,0 +1,212 @@
+/* (c) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.source.extractor.extract.jdbc;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Types;
+import java.util.Iterator;
+import java.util.List;
+
+import org.testng.annotations.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import gobblin.configuration.State;
+import gobblin.configuration.WorkUnitState;
+import gobblin.source.extractor.extract.CommandOutput;
+import gobblin.source.extractor.extract.jdbc.JdbcCommand;
+import gobblin.source.extractor.extract.jdbc.JdbcCommandOutput;
+import gobblin.source.extractor.extract.jdbc.JdbcExtractor;
+import gobblin.source.extractor.extract.jdbc.MysqlExtractor;
+
+@Test(groups = { "gobblin.source.extractor.extract.jdbc" })
+public class JdbcExtractorTest {
+
+  private final static List<MockJdbcColumn> COLUMNS = ImmutableList.of(new MockJdbcColumn("id", "1", Types.INTEGER),
+      new MockJdbcColumn("name", "name_1", Types.VARCHAR), new MockJdbcColumn("age", "20", Types.INTEGER));
+
+  @Test
+  public void testGetData() throws Exception {
+
+    CommandOutput<JdbcCommand, ResultSet> output = new JdbcCommandOutput();
+    output.put(new JdbcCommand(), buildMockResultSet());
+
+    State state = new WorkUnitState();
+    state.setId("id");
+    JdbcExtractor jdbcExtractor = new MysqlExtractor((WorkUnitState) state);
+
+    List<String> columnNames = Lists.newArrayListWithCapacity(COLUMNS.size());
+
+    for (MockJdbcColumn mockJdbcColumn:COLUMNS) {
+      columnNames.add(mockJdbcColumn.getColumnName());
+    }
+
+    jdbcExtractor.setHeaderRecord(columnNames);
+
+    Iterator<JsonElement> itr = jdbcExtractor.getData(output);
+
+    // Make sure there is an element in the iterator
+    assertTrue(itr.hasNext());
+
+    JsonObject obj = itr.next().getAsJsonObject();
+
+    // Verify the columns
+    for (MockJdbcColumn column : COLUMNS) {
+      assertEquals(obj.get(column.getColumnName()).getAsString(), column.getValue());
+    }
+  }
+
+  /*
+   * Build a mock implementation of Result using Mockito
+   */
+  private ResultSet buildMockResultSet() throws Exception {
```

Comment:
Have you looked at MockResultSet in https://github.com/mockrunner/mockrunner ? It is a bit more generic way to mock result sets and it can deal with multiple records.